### ISSUE TITLE
On the definition of groups

### ISSUE THREAD

**Issue by mikeshulman** [HoTT/HoTT#661]
A radical suggestion:

``` coq
Record Group :=
  { classifying_space : Type ;
    grpt : classifying_space ;
    isconn_classifying_space : IsConnected 0 classifying_space ;
    istrunc_classifying_space : IsTrunc 1 classifying_space }.

Definition group_set (G : Group) : Type
  := grpt G = grpt G.

Coercion group_set : Group >-> Sortclass.
```

**Comment by spitters** [comment]
@andrejbauer In general, I agree with your dislike for particular implementations.
About the sayings. It's [you can't have your cake and eat it](http://en.wikipedia.org/wiki/You_can%27t_have_your_cake_and_eat_it) in English. We say: to save the goat and the cabbage. Here's the [xkcd solution to the problem](http://xkcd.com/1134/).

**Comment by Alizter** [comment]
Has this been solved by #1122? And since we are implementing classifying spaces soon, we can also show that every group is isomorpic to `loops (B G)` which means we can use the advantages of Mike's suggestion whenever.

**Comment by mikeshulman** [comment]
Yes, I guess we've decided to use hottclasses groups to fit them into the algebraic hierarchy.
